TriLabelView/TriLabelView.swift

```diff
@@ -15,198 +15,218 @@ import UIKit
-      labelAngle = (3.14/4)
-    case .BottomRight:
-      x = (2/3*rectWidth+rectOriginX)-textWidth/2
-      y = (2/3*rectWidth+rectOriginY)-textHeight/2
-      labelAngle = (-3.14/4)
-    case .BottomLeft:
-      x = (1/3*rectWidth+rectOriginX)-textWidth/2
-      y = (2/3*rectWidth+rectOriginY)-textHeight/2
-      labelAngle = (3.14/4)
-    default:
-      x = (1/3*rectWidth+rectOriginX) - textWidth/2
-      y = (1/3*rectWidth+rectOriginY) - textHeight/2
-      labelAngle = (7*CGFloat(3.14/4))
-    }
-    return(x,y,labelAngle,textWidth,textHeight)
-  }
-  
-  /// Get the width and height of the text.
-  private func getTextCGSize(_ text:String) -> (CGFloat,CGFloat) {
-    let uiFont:UIFont = UIFont.init(name: "HelveticaNeue-Bold", size: fontSize)!
-    let textAttr = [NSFontAttributeName:uiFont]
-    let nsText = text as NSString
-    let cgSize = nsText.size(attributes: textAttr)
-    return (cgSize.width,cgSize.height)
-  }
-  
-}
-
-/// The positions to select for placement of the label view
-public enum Position:String {
-  case TopLeft = "topleft"
-  case TopRight = "topright"
-  case BottomRight = "bottomright"
-  case BottomLeft = "bottomleft"
+    
+    /// The positions to select for placement of the label view.
+    public enum Position:String {
+        case TopLeft = "topleft"
+        case TopRight = "topright"
+        case BottomRight = "bottomright"
+        case BottomLeft = "bottomleft"
+    }
+    
+    private var length = CGFloat()
+    private var newRect = CGRect()
+    
+    // MARK: - Initialization
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setUp()
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setUp()
+    }
+    
+    private func setUp() {
+        self.isOpaque = false
+    }
+    
+    /// The position of the label view.
+    public var position:Position = .TopLeft {
+        didSet {
+            setNeedsDisplay()
+        }
+    }
+    
+    // MARK: - Attributes
+    
+    /// The postion of the label view for IB.
+    @available(*, unavailable, message: "This property is reserved for IB. Use position instead")
+    @IBInspectable public var positionName: String? {
+        didSet {
+            setNeedsDisplay()
+        }
+        willSet {
+            if let newPosition = Position(rawValue: newValue?.lowercased() ?? "") {
+                position = newPosition
+            }
+        }
+    }
+    
+    /// Percentage of length of the label view to parent view.
+    @IBInspectable public var lengthPercentage:CGFloat = 50.0 {
+        didSet {
+            setNeedsDisplay()
+        }
+    }
+    
+    /// The text displayed by the label view.
+    @IBInspectable public var labelText:String = "Hi" {
+        didSet {
+            setNeedsDisplay()
+        }
+    }
+    
+    /// The font size of the text.
+    @IBInspectable public var fontSize:CGFloat = 20
```

Comment:
We can remove this property as font size can be passed in the font itself and after your change, we are not using `fontSize` anywhere in the code

Reply:
Oops. Must've left that one in there. Let me fix it.